src/apps/DIFFUSION3DPA-Hip.cpp

```diff
@@ -0,0 +1,288 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-21, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DIFFUSION3DPA.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_HIP)
+
+#include "common/HipDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf {
+namespace apps {
+
+#define DIFFUSION3DPA_DATA_SETUP_HIP                                        \
+  allocAndInitHipDeviceData(Basis, m_B, DPA_Q1D *DPA_D1D);                  \
+  allocAndInitHipDeviceData(dBasis, m_G, DPA_Q1D *DPA_D1D);                 \
+  allocAndInitHipDeviceData(D, m_D, DPA_Q1D *DPA_Q1D *DPA_Q1D *SYM *m_NE);  \
+  allocAndInitHipDeviceData(X, m_X, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);       \
+  allocAndInitHipDeviceData(Y, m_Y, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);
+
+#define DIFFUSION3DPA_DATA_TEARDOWN_HIP                                   \
+  getHipDeviceData(m_Y, Y, DPA_D1D *DPA_D1D *DPA_D1D *m_NE);              \
+  deallocHipDeviceData(Basis);                                            \
+  deallocHipDeviceData(dBasis);                                           \
+  deallocHipDeviceData(D);                                                \
+  deallocHipDeviceData(X);                                                \
+  deallocHipDeviceData(Y);
+
+// Uncomment to add compiler directives for loop unrolling
+//#define USE_RAJA_UNROLL
```

Comment:
Why is this still here, wouldn't it need to be above the include to work?

Reply:
hey good catch! sure let me move up.